<!DOCTYPE HTML>
<html lang="en" class="rust" dir="ltr">
    <head>
        <!-- Book generated using mdBook -->
        <meta charset="UTF-8">
        <title>Transcriptomics I - Notes @ ETH Zürich</title>


        <!-- Custom HTML head -->
        
        <meta name="description" content="">
        <meta name="viewport" content="width=device-width, initial-scale=1">
        <meta name="theme-color" content="#ffffff">

        <link rel="icon" href="../../favicon.svg">
        <link rel="shortcut icon" href="../../favicon.png">
        <link rel="stylesheet" href="../../css/variables.css">
        <link rel="stylesheet" href="../../css/general.css">
        <link rel="stylesheet" href="../../css/chrome.css">
        <link rel="stylesheet" href="../../css/print.css" media="print">

        <!-- Fonts -->
        <link rel="stylesheet" href="../../FontAwesome/css/font-awesome.css">
        <link rel="stylesheet" href="../../fonts/fonts.css">

        <!-- Highlight.js Stylesheets -->
        <link rel="stylesheet" href="../../highlight.css">
        <link rel="stylesheet" href="../../tomorrow-night.css">
        <link rel="stylesheet" href="../../ayu-highlight.css">

        <!-- Custom theme stylesheets -->
        <link rel="stylesheet" href="../../theme/custom.css">

        <!-- MathJax -->
        <script async src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    </head>
    <body class="sidebar-visible no-js">
    <div id="body-container">
        <!-- Provide site root to javascript -->
        <script>
            var path_to_root = "../../";
            var default_theme = window.matchMedia("(prefers-color-scheme: dark)").matches ? "ayu" : "rust";
        </script>

        <!-- Work around some values being stored in localStorage wrapped in quotes -->
        <script>
            try {
                var theme = localStorage.getItem('mdbook-theme');
                var sidebar = localStorage.getItem('mdbook-sidebar');

                if (theme.startsWith('"') && theme.endsWith('"')) {
                    localStorage.setItem('mdbook-theme', theme.slice(1, theme.length - 1));
                }

                if (sidebar.startsWith('"') && sidebar.endsWith('"')) {
                    localStorage.setItem('mdbook-sidebar', sidebar.slice(1, sidebar.length - 1));
                }
            } catch (e) { }
        </script>

        <!-- Set the theme before any content is loaded, prevents flash -->
        <script>
            var theme;
            try { theme = localStorage.getItem('mdbook-theme'); } catch(e) { }
            if (theme === null || theme === undefined) { theme = default_theme; }
            var html = document.querySelector('html');
            html.classList.remove('rust')
            html.classList.add(theme);
            var body = document.querySelector('body');
            body.classList.remove('no-js')
            body.classList.add('js');
        </script>

        <input type="checkbox" id="sidebar-toggle-anchor" class="hidden">

        <!-- Hide / unhide sidebar before it is displayed -->
        <script>
            var body = document.querySelector('body');
            var sidebar = null;
            var sidebar_toggle = document.getElementById("sidebar-toggle-anchor");
            if (document.body.clientWidth >= 1080) {
                try { sidebar = localStorage.getItem('mdbook-sidebar'); } catch(e) { }
                sidebar = sidebar || 'visible';
            } else {
                sidebar = 'hidden';
            }
            sidebar_toggle.checked = sidebar === 'visible';
            body.classList.remove('sidebar-visible');
            body.classList.add("sidebar-" + sidebar);
        </script>

        <nav id="sidebar" class="sidebar" aria-label="Table of contents">
            <div class="sidebar-scrollbox">
                <ol class="chapter"><li class="chapter-item affix "><a href="../../index.html">Home</a></li><li class="spacer"></li><li class="chapter-item affix "><li class="part-title">22HS</li><li class="chapter-item "><div><strong aria-hidden="true">1.</strong> Computational Biology</div></li><li class="chapter-item "><a href="../../22hs/dm1/data_mining_i.html"><strong aria-hidden="true">2.</strong> Data Mining I</a><a class="toggle"><div>❱</div></a></li><li><ol class="section"><li class="chapter-item "><a href="../../22hs/dm1/01_metrics.html"><strong aria-hidden="true">2.1.</strong> Metrics</a></li><li class="chapter-item "><a href="../../22hs/dm1/02_classification.html"><strong aria-hidden="true">2.2.</strong> Classification</a></li><li class="chapter-item "><a href="../../22hs/dm1/03_clustering.html"><strong aria-hidden="true">2.3.</strong> Clustering</a></li><li class="chapter-item "><a href="../../22hs/dm1/04_feature_selection.html"><strong aria-hidden="true">2.4.</strong> Feature Selection</a></li><li class="chapter-item "><a href="../../22hs/dm1/05_applications_in_computational_biology.html"><strong aria-hidden="true">2.5.</strong> Applications in Computational Biology</a></li><li class="spacer"></li></ol></li><li class="chapter-item "><li class="part-title">23FS + HS</li><li class="chapter-item "><a href="../../23fs/mham/mobile_health_and_activity_monitoring.html"><strong aria-hidden="true">3.</strong> Mobile Health and Activity Monitoring</a><a class="toggle"><div>❱</div></a></li><li><ol class="section"><li class="chapter-item "><div><strong aria-hidden="true">3.1.</strong> Introduction</div></li></ol></li><li class="chapter-item "><a href="../../23fs/bdfe/big_data_for_engineers.html"><strong aria-hidden="true">4.</strong> Big Data for Engineers</a><a class="toggle"><div>❱</div></a></li><li><ol class="section"><li class="chapter-item "><a href="../../23fs/bdfe/01_azure_blob_storage.html"><strong aria-hidden="true">4.1.</strong> Azure Blob Storage</a></li><li class="chapter-item "><a href="../../23fs/bdfe/02_hadoop.html"><strong aria-hidden="true">4.2.</strong> Hadoop</a></li><li class="chapter-item "><a href="../../23fs/bdfe/03_spark.html"><strong aria-hidden="true">4.3.</strong> Spark</a></li><li class="chapter-item "><a href="../../23fs/bdfe/04_jsoniq.html"><strong aria-hidden="true">4.4.</strong> JSONiq</a></li><li class="chapter-item "><a href="../../23fs/bdfe/05_hbase_(wide_column_store).html"><strong aria-hidden="true">4.5.</strong> HBase (Wide Column Store)</a></li><li class="chapter-item "><a href="../../23fs/bdfe/06_mapreduce.html"><strong aria-hidden="true">4.6.</strong> MapReduce</a></li><li class="chapter-item "><a href="../../23fs/bdfe/07_yarn.html"><strong aria-hidden="true">4.7.</strong> YARN</a></li></ol></li><li class="chapter-item expanded "><a href="../../23fs/fg/functional_genomics.html"><strong aria-hidden="true">5.</strong> Functional Genomics</a><a class="toggle"><div>❱</div></a></li><li><ol class="section"><li class="chapter-item "><a href="../../23fs/fg/01_modern_genomics_i.html"><strong aria-hidden="true">5.1.</strong> Modern Genomics I</a></li><li class="chapter-item "><a href="../../23fs/fg/02_modern_genomics_ii.html"><strong aria-hidden="true">5.2.</strong> Modern Genomics II</a></li><li class="chapter-item expanded "><a href="../../23fs/fg/03_transcriptomics_i.html" class="active"><strong aria-hidden="true">5.3.</strong> Transcriptomics I</a></li><li class="chapter-item "><a href="../../23fs/fg/04_transcriptomics_ii.html"><strong aria-hidden="true">5.4.</strong> Transcriptomics II</a></li><li class="chapter-item "><div><strong aria-hidden="true">5.5.</strong> MicroRNAs and other small RNAs</div></li><li class="chapter-item "><div><strong aria-hidden="true">5.6.</strong> Proteomics</div></li><li class="chapter-item "><div><strong aria-hidden="true">5.7.</strong> Metabolomics</div></li><li class="chapter-item "><div><strong aria-hidden="true">5.8.</strong> Single Cell Mass Cytometry</div></li><li class="chapter-item "><div><strong aria-hidden="true">5.9.</strong> Protein Networks</div></li><li class="chapter-item "><div><strong aria-hidden="true">5.10.</strong> Epigenomics and Gene Regulation</div></li><li class="chapter-item "><div><strong aria-hidden="true">5.11.</strong> Qualitiy Control and Standards</div></li><li class="spacer"></li></ol></li><li class="chapter-item "><li class="part-title">24FS</li><li class="chapter-item "><div><strong aria-hidden="true">6.</strong> Biofluiddynamics</div></li><li class="chapter-item "><div><strong aria-hidden="true">7.</strong> Synthetic Biology I</div></li><li class="chapter-item affix "><li class="part-title">24HS</li><li class="chapter-item "><div><strong aria-hidden="true">8.</strong> Computer Architecture</div></li><li class="chapter-item "><div><strong aria-hidden="true">9.</strong> Applied Bioinformatics: Microbiomes</div></li><li class="chapter-item "><a href="../../24hs/evodynamo/evolutionary_dynamics.html"><strong aria-hidden="true">10.</strong> Evolutionary Dynamics</a><a class="toggle"><div>❱</div></a></li><li><ol class="section"><li class="chapter-item "><a href="../../24hs/evodynamo/01_what_is_evolution.html"><strong aria-hidden="true">10.1.</strong> What is Evolution?</a></li><li class="chapter-item "><a href="../../24hs/evodynamo/02_quasispecies.html"><strong aria-hidden="true">10.2.</strong> Quasispecies</a></li><li class="chapter-item "><a href="../../24hs/evodynamo/03_stochastic_modles_of_finite_populations.html"><strong aria-hidden="true">10.3.</strong> Stochastic Models of Finite Populations</a></li><li class="chapter-item "><a href="../../24hs/evodynamo/04_evolutionary_dynamics_of_cancer.html"><strong aria-hidden="true">10.4.</strong> Evolutionary Dynamics of Cancer</a></li><li class="chapter-item "><div><strong aria-hidden="true">10.5.</strong> Cancer Progression: The Speed of Adaptation</div></li><li class="chapter-item "><div><strong aria-hidden="true">10.6.</strong> Diffusion Theory</div></li><li class="chapter-item "><div><strong aria-hidden="true">10.7.</strong> Evolutionary Game Theory</div></li><li class="chapter-item "><div><strong aria-hidden="true">10.8.</strong> Evolutionary Games in Finite Populations</div></li><li class="chapter-item "><div><strong aria-hidden="true">10.9.</strong> Spatial Models for the Evolution of Solid Tumors</div></li><li class="chapter-item "><div><strong aria-hidden="true">10.10.</strong> Branching Processes</div></li><li class="chapter-item "><div><strong aria-hidden="true">10.11.</strong> Evolutionary Escape</div></li><li class="chapter-item "><div><strong aria-hidden="true">10.12.</strong> Coalescent Theory</div></li><li class="chapter-item "><div><strong aria-hidden="true">10.13.</strong> Tumor Archeology</div></li></ol></li></ol>
            </div>
            <div id="sidebar-resize-handle" class="sidebar-resize-handle">
                <div class="sidebar-resize-indicator"></div>
            </div>
        </nav>

        <!-- Track and set sidebar scroll position -->
        <script>
            var sidebarScrollbox = document.querySelector('#sidebar .sidebar-scrollbox');
            sidebarScrollbox.addEventListener('click', function(e) {
                if (e.target.tagName === 'A') {
                    sessionStorage.setItem('sidebar-scroll', sidebarScrollbox.scrollTop);
                }
            }, { passive: true });
            var sidebarScrollTop = sessionStorage.getItem('sidebar-scroll');
            sessionStorage.removeItem('sidebar-scroll');
            if (sidebarScrollTop) {
                // preserve sidebar scroll position when navigating via links within sidebar
                sidebarScrollbox.scrollTop = sidebarScrollTop;
            } else {
                // scroll sidebar to current active section when navigating via "next/previous chapter" buttons
                var activeSection = document.querySelector('#sidebar .active');
                if (activeSection) {
                    activeSection.scrollIntoView({ block: 'center' });
                }
            }
        </script>

        <div id="page-wrapper" class="page-wrapper">

            <div class="page">
                                <div id="menu-bar-hover-placeholder"></div>
                <div id="menu-bar" class="menu-bar sticky">
                    <div class="left-buttons">
                        <label id="sidebar-toggle" class="icon-button" for="sidebar-toggle-anchor" title="Toggle Table of Contents" aria-label="Toggle Table of Contents" aria-controls="sidebar">
                            <i class="fa fa-bars"></i>
                        </label>
                        <button id="theme-toggle" class="icon-button" type="button" title="Change theme" aria-label="Change theme" aria-haspopup="true" aria-expanded="false" aria-controls="theme-list">
                            <i class="fa fa-paint-brush"></i>
                        </button>
                        <ul id="theme-list" class="theme-popup" aria-label="Themes" role="menu">
                            <li role="none"><button role="menuitem" class="theme" id="light">Light</button></li>
                            <li role="none"><button role="menuitem" class="theme" id="rust">Rust</button></li>
                            <li role="none"><button role="menuitem" class="theme" id="coal">Coal</button></li>
                            <li role="none"><button role="menuitem" class="theme" id="navy">Navy</button></li>
                            <li role="none"><button role="menuitem" class="theme" id="ayu">Ayu</button></li>
                        </ul>
                        <button id="search-toggle" class="icon-button" type="button" title="Search. (Shortkey: s)" aria-label="Toggle Searchbar" aria-expanded="false" aria-keyshortcuts="S" aria-controls="searchbar">
                            <i class="fa fa-search"></i>
                        </button>
                    </div>

                    <h1 class="menu-title">Notes @ ETH Zürich</h1>

                    <div class="right-buttons">
                        <a href="../../print.html" title="Print this book" aria-label="Print this book">
                            <i id="print-button" class="fa fa-print"></i>
                        </a>

                    </div>
                </div>

                <div id="search-wrapper" class="hidden">
                    <form id="searchbar-outer" class="searchbar-outer">
                        <input type="search" id="searchbar" name="searchbar" placeholder="Search this book ..." aria-controls="searchresults-outer" aria-describedby="searchresults-header">
                    </form>
                    <div id="searchresults-outer" class="searchresults-outer hidden">
                        <div id="searchresults-header" class="searchresults-header"></div>
                        <ul id="searchresults">
                        </ul>
                    </div>
                </div>

                <!-- Apply ARIA attributes after the sidebar and the sidebar toggle button are added to the DOM -->
                <script>
                    document.getElementById('sidebar-toggle').setAttribute('aria-expanded', sidebar === 'visible');
                    document.getElementById('sidebar').setAttribute('aria-hidden', sidebar !== 'visible');
                    Array.from(document.querySelectorAll('#sidebar a')).forEach(function(link) {
                        link.setAttribute('tabIndex', sidebar === 'visible' ? 0 : -1);
                    });
                </script>

                <div id="content" class="content">
                    <main>
                        <h1 id="transcriptomics-i"><a class="header" href="#transcriptomics-i">Transcriptomics I</a></h1>
<p>Unimportant text is grayed out but worth reading for fluency and context.</p>
<ul>
<li><a href="#motivation">Motivation</a></li>
<li><a href="#rna-species">RNA species</a></li>
<li><a href="#rna-selection-methods">RNA selection methods</a></li>
<li><a href="#rna-seq-experiment-workflow-overview">RNA-seq experiment workflow overview</a></li>
<li><a href="#rna-seq-experiment-design">RNA-seq experiment design</a></li>
<li><a href="#rna-seq-read-mapping-methods">RNA-seq read mapping methods</a>
<ul>
<li><a href="#considerations">Considerations</a>
<ul>
<li><a href="#how-to-map-reads-to-a-genome">How to map reads to a genome?</a></li>
<li><a href="#what-if-we-want-to-map-billions-of-reads">What if we want to map billions of reads?</a></li>
<li><a href="#how-to-deal-with-snps-and-indels">How to deal with SNPs and indels?</a></li>
<li><a href="#how-to-deal-with-introns-basically-very-large-insertions">How to deal with introns (basically very large insertions)?</a></li>
</ul>
</li>
<li><a href="#bowtie">Bowtie</a></li>
<li><a href="#tophat2">TopHat2</a></li>
<li><a href="#star">STAR</a></li>
</ul>
</li>
<li><a href="#quality-control-and-reporting-considerations-after-mapping">Quality control and reporting considerations after mapping</a>
<ul>
<li><a href="#phread-quality">PHREAD quality</a></li>
<li><a href="#read-trimming">Read trimming</a></li>
<li><a href="#multiple-alignments-and-report-considerations">Multiple alignments and report considerations</a></li>
</ul>
</li>
<li><a href="#expression-quantification">Expression quantification</a>
<ul>
<li><a href="#how-to-count-reads">How to count reads?</a>
<ul>
<li><a href="#em-estimation">EM estimation</a></li>
<li><a href="#alignment-free-rna-seq-quantification">Alignment-free RNA-seq quantification</a></li>
</ul>
</li>
<li><a href="#normalization">Normalization</a></li>
</ul>
</li>
</ul>
<h2 id="motivation"><a class="header" href="#motivation">Motivation</a></h2>
<p><span style="color:gray"> We are interested in the expression of genes in a
cell. Altough <u>proteins are the actors</u> of genes instead of mRNA
transcripts, <em>mRNAs transcripts are easier to measure</em> because selection
has chosen them to convey the genetic message in a faithful way. Also there are
plenty of interesting <em>non-coding RNAs</em>. </span></p>
<h2 id="rna-species"><a class="header" href="#rna-species">RNA species</a></h2>
<p>80% of the total RNA is rRNA. 14% is tRNA. Only 1-3% is mRNA. All other species
account for the remaining 1-3%.</p>
<p><img src="img/03-rna-species.png" alt="RNA species" style="zoom:50%;" /><img
src="img/03-rna-prevalence-pie-chart.png" alt="Prevalence of RNA species"
style="zoom:50%;" /></p>
<h2 id="rna-selection-methods"><a class="header" href="#rna-selection-methods">RNA selection methods</a></h2>
<p><img src="img/03-rna-selection-method.png" alt="RNA selection methods" /></p>
<ul>
<li>A: total RNA
<ul>
<li>Unbiased estimation</li>
<li>Dominate by rRNA, not very informative</li>
</ul>
</li>
<li>B: ribosomal reduction/depletion
<ul>
<li>Use probes that bind to rRNA and pull them out</li>
<li>&lt; 5% rRNA left</li>
</ul>
</li>
<li>C: poly-A selection
<ul>
<li>Use probes that bind to poly-A tails</li>
<li>Selects for mRNA, and also some long non-coding RNAs (because they also have
poly-A tails)</li>
</ul>
</li>
<li>D: cDNA capture
<ul>
<li>Design target probes that bind to a certain set of RNAs</li>
<li>Limited transcriptome coverage</li>
</ul>
</li>
</ul>
<p><span style="color:gray"> 90% of transcriptome research utilize poly-A
selection; 5% of them go for ribosomal depletion because their mRNAs are of low
quality and fragmented. The rest 5% do have high-quality mRNAs but still go for
ribosomal depletion because they want to estimate other RNA species at the same
time.</span></p>
<p>Note: current illumina sequencers cannot sequence miRNAs (20-25nt) and mRNA
fragments (100-300nt) together (<u>because of the large length discrepency</u>)</p>
<h2 id="rna-seq-experiment-workflow-overview"><a class="header" href="#rna-seq-experiment-workflow-overview">RNA-seq experiment workflow overview</a></h2>
<p>A typical RNA-seq experiment workflow is shown below.</p>
<p><img src="img/03-rna-seq-experiment-workflow.png" alt="Typical RNA-seq experiment" /></p>
<p><span style="color:gray"> We can do either single-read sequencing or paired-end
sequencing (as is shown in the figure above) in the third step. </span></p>
<p><img src="img/03-paired-end-seq.png" alt="Paired-end sequencing"
style="zoom:60%;" /></p>
<p>It can also be summarized into a flow diagram.</p>
<p><img src="img/03-ran-seq-workflow.png" alt="RNA-seq workflow diagram"
style="zoom:67%;" /></p>
<p>The following sections will go into more details about each step.</p>
<h2 id="rna-seq-experiment-design"><a class="header" href="#rna-seq-experiment-design">RNA-seq experiment design</a></h2>
<ul>
<li>RNA-seq experiments are implemented as <strong>comparative experiments</strong></li>
<li>RNA-seq is used to measure the <strong>relative abundance (expression)</strong> of each
mRNA (gene) in a <u>sample</u>
<ul>
<li><span style="color:gray"><u>sample</u></span><span style="color:gray">: We
cannot sequence all the RNA fragments. Instead, we randomly select about 20M
fragments as a representation of the total RNA obtained. Note that we can
sample multiple times. </span></li>
</ul>
</li>
<li><span style="color:gray">RNA-seq can link gene expression to</span>
<ul>
<li><span style="color:gray">genotype</span></li>
<li><span style="color:gray">phenotype</span></li>
<li><span style="color:gray">treatment response </span></li>
</ul>
</li>
</ul>
<h2 id="rna-seq-read-mapping-methods"><a class="header" href="#rna-seq-read-mapping-methods">RNA-seq read mapping methods</a></h2>
<h3 id="considerations"><a class="header" href="#considerations">Considerations</a></h3>
<p><span style="color:gray"> Ideally, the read mapping process is as simple as
<strong>sequencing the transcribed RNAs</strong> and then <strong>mapping the
sequences back to the genome</strong> to identify the gene expression. </span></p>
<p><span style="color:gray"> However, in reality, the process is much more
complicated because <u>current sequencers can only perform faithful sequencing
up to a couple of hundreds of bp (usually 200-800bp)</u>, and a typical mRNA is
2-3kb long. Thus, the mRNA needs to be first chopped into fragments and then
sequenced. When we do the mapping, we're mapping the chopped reads back to the
genome instead of the whole spliced sequences. The existence of <u>alternative
splicing (protein isoforms)</u> and <u>sequencing errors</u> makes the mapping
process even more complicated. </span></p>
<p><span style="color:gray"> While discussing the mapping process, those
limitations and problems should be kept in mind. </span></p>
<h4 id="how-to-map-reads-to-a-genome"><a class="header" href="#how-to-map-reads-to-a-genome">How to map reads to a genome?</a></h4>
<ul>
<li>Mapping algorithms must be
<ul>
<li>Fast
<ul>
<li>Thus usually find perfect or near-perfect (tolerating 1bp mismatch) match</li>
</ul>
</li>
<li>Able to handle SNPs, indels and sequencing errors</li>
</ul>
</li>
</ul>
<p>We can always perform sequence alignment (here needs local
alignment/<strong>Smith-Waterman alignment</strong>). Of course the score matrix should be
good to determine the correct alignment. But this is slow because it requires
quadratic efforts.</p>
<h4 id="what-if-we-want-to-map-billions-of-reads"><a class="header" href="#what-if-we-want-to-map-billions-of-reads">What if we want to map billions of reads?</a></h4>
<ul>
<li><strong>Indexing method</strong>
<ul>
<li>Reads are aligned by <strong>index lookup</strong> instead of $O(n^2)$ seq comparison</li>
<li>Usually hashed so the lookup time is much faster</li>
<li>If lookup fails, actual sequence comparison is performed (more details on
this in next chapter)</li>
<li>Major aligners use the <u><strong>Burrows-Wheeler transform (BWT)</strong></u> to index
the reference genome
<ul>
<li><span style="color:gray">Very small. Even for the human genome the index
fits into 3GB RAM. </span></li>
</ul>
</li>
</ul>
</li>
<li>Example aligners
<ul>
<li><a href="#bowtie"><strong>bowtie</strong>, <strong>bowtie2</strong></a></li>
<li>BWA</li>
<li><a href="#tophat2"><strong>TopHat2</strong></a></li>
<li><a href="#star"><strong>STAR</strong></a></li>
<li>SOAP</li>
<li>...</li>
</ul>
</li>
</ul>
<h4 id="how-to-deal-with-snps-and-indels"><a class="header" href="#how-to-deal-with-snps-and-indels">How to deal with SNPs and indels?</a></h4>
<ul>
<li><strong>SNPs</strong>
<ul>
<li><span style="color:gray">Can always be handled by performing real
Smith-Waterman alignment but expensive </span></li>
<li>At mismatched positions, try all possible bases and search the BWT index
again
<ul>
<li>Computing effort grows exponentially</li>
</ul>
</li>
<li><strong>Gapped alignment</strong>
<ul>
<li>Chop reads into short segments (seeds)</li>
<li>align those seeds in a <u>mismatch-free</u> manner (typically using BWT
index lookup again)</li>
<li>stitch them back together</li>
<li>Might require multiple rounds of chopping</li>
</ul>
</li>
</ul>
</li>
<li><strong>Indels</strong>
<ul>
<li>Can only be found with <strong>gapped alignment</strong> (described above), BWT lookup is
not enough</li>
</ul>
</li>
</ul>
<h4 id="how-to-deal-with-introns-basically-very-large-insertions"><a class="header" href="#how-to-deal-with-introns-basically-very-large-insertions">How to deal with introns (basically very large insertions)?</a></h4>
<ul>
<li>Approaches
<ul>
<li>Map directly to <strong>trapscript sequences</strong> (no intron) not to the genome
<ul>
<li>Pro: introns are not a problem</li>
<li>Cons: unknown genes/isoforms cannot be detected</li>
</ul>
</li>
<li><strong>Spliced alignment</strong> to genome
<ul>
<li>Pro: finds reads from unknown gene loci or unknown isoforms</li>
<li>Con: larger search space, potentially more false positives, wrong
alignment to <u>pseudogenes</u></li>
</ul>
</li>
<li>Combination of above</li>
</ul>
</li>
<li>Difference between spliced alignment and gapped alignment
<ul>
<li>Gapped alignment: longer gaps means high score penalty and lower score</li>
<li>Spliced alignment: long gaps at canonical splice-sites <u>are allowed</u>,
but lacking canonical splice-sites or gaps elsewhere (at non-canonical
splice sites) are penalized</li>
</ul>
</li>
<li>Pseudogenes
<ul>
<li>associated <em>retrotransposed</em> spliced sequence of a certain gene, thus
<em>intron-free</em>, and appears &quot;<em>exonic</em>&quot;</li>
<li>Have almost the same sequence as the original gene</li>
<li>Not functional because they do not have access to promotors so won't be
transcribed</li>
<li>Usually contain more mutations because there are no selection pressures to
keep them intact</li>
<li>Exonic reads will map to both the gene and its pseudogene, but likely
preferring gene placement due to lack of mutations</li>
<li>A spliced read could be incorrectly assigned to the pseudogene as it appears
to be exonic, preventing higher scoring spliced alignments from being
pursued</li>
</ul>
</li>
</ul>
<h3 id="bowtie"><a class="header" href="#bowtie">Bowtie</a></h3>
<p><img src="img/03-bowtie.png" alt="Bowtie" /></p>
<ul>
<li>Step-by-step backward searching the <u><strong>suffix</strong></u> of the query sequence in
the BWT index
<ul>
<li>Exact matching</li>
</ul>
</li>
<li>Searched suffix appears consecutively in BWT</li>
<li>The size of range in BWT shrinks or remains the same</li>
<li><u><strong>almost does not handle mismatches</strong></u>; a single mismatch will lead to
<u>empty BWT</u> range / failed index lookup
<ul>
<li>mismatches can come from
<ul>
<li>Sequencing error (illumina: $1/1000$)</li>
<li>True variation (SNPs, human mutation rate $\approx 1/10^4$</li>
</ul>
</li>
<li>mismatches are not rare events! at least 10% of &gt;100nt reads</li>
</ul>
</li>
<li>Empty BWT range activates <strong>backtracking</strong>
<ul>
<li>All possible bases are tried at the mismatched position</li>
<li>Gapped alignment</li>
</ul>
</li>
</ul>
<h3 id="tophat2"><a class="header" href="#tophat2">TopHat2</a></h3>
<p><strong>Handles mismatches</strong> by gapped alignment. It <u>first</u> performs (optional)
<u>transcriptome alignment</u> (exonic), and then performs <u>genome
alignment</u> for the unmapped reads. The unmapped reads in the second step are
multi-exon spanning reads and undergo <u>spliced alignment</u> by chopping the
reads into even shorter segments and splice site identification etc.</p>
<blockquote>
<p>The figure is too complicated, not recommended to go into details but still
attached here for reference. <img src="img/03-tophat2.png" alt="TopHat2 pipeline" /></p>
</blockquote>
<h3 id="star"><a class="header" href="#star">STAR</a></h3>
<p><strong>Direct</strong> alignment of non-contiguous sequences to the reference genome.</p>
<ul>
<li>First find Maximal Mappable Prefix (MMP) of the read
<ul>
<li>MMP is the longest prefix of the read that can be mapped to the reference
genome</li>
<li>MMP is found by searching the BWT index</li>
<li>Possible to tolerate a certain degree of mismatches using extend mode</li>
</ul>
</li>
<li>Stop extending MMP if MMP covers the whole read or encounters a mismatch</li>
<li>In the second case, find the MMP of the remaining suffix of the read</li>
<li>Repeat until the whole read is covered
<ul>
<li>Possible to use trim mode to get rid of A-tail or adapter sequence</li>
</ul>
</li>
</ul>
<img src="img/03-star.png" alt="STAR" style="zoom:67%;" />
<h2 id="quality-control-and-reporting-considerations-after-mapping"><a class="header" href="#quality-control-and-reporting-considerations-after-mapping">Quality control and reporting considerations after mapping</a></h2>
<h3 id="phread-quality"><a class="header" href="#phread-quality">PHREAD quality</a></h3>
<ul>
<li>Each called base is given a quality score $Q$</li>
<li>$Q = -10 \log_{10}(p)$ (PHRED score)
<ul>
<li>$p$ is the estimated probability of the base being called incorrectly</li>
<li>$Q$ is the negative log of the probability
<ul>
<li>Q-score = 10: prob = 0.1</li>
<li>Q-score = 20: prob = 0.01</li>
<li>Q-score = 30: prob = 0.001</li>
</ul>
</li>
<li><strong>Q30</strong> is often used as a cutoff for high quality reads</li>
</ul>
</li>
<li>How to use: downweight the low-quality bases when computing the alignment
score</li>
<li><span style="color:gray">First introduced for capillary sequencers</span></li>
<li><span style="color:gray">PHRED scores are determined by the sequencer that
directly rates how reliable the measured signal is</span></li>
</ul>
<h3 id="read-trimming"><a class="header" href="#read-trimming">Read trimming</a></h3>
<p>Not all parts of the read are useful for alignment. It is common to trim the
reads.</p>
<ul>
<li>Reasons for trimming:
<ul>
<li>Systemetic errors of sequencer
<ul>
<li>Illumina sequencers have high error rate at the first few bases</li>
<li>Basically all sequencers have increasing error rate towards the end of the
read</li>
</ul>
</li>
<li>Adapter trimming
<ul>
<li>If the inserted DNA/RNA fragment is too short, the read will contain part
of the adapter (since the sequencer generally will sequence 100bp at each
end)</li>
<li>Can be challenging if:
<ul>
<li>The inserted sequence is 90-100bp -- the incorporated adapter sequence
will only be a few bp long</li>
<li>The read has many sequencing errors</li>
</ul>
</li>
</ul>
</li>
<li>3'-bias / degradation
<ul>
<li>The 3' end of the read is generally more likely to be degraded than the 5'
end</li>
<li>Depending on RNA degradation and extration protocols reads may not be
equally distributed along the transcripts</li>
</ul>
</li>
</ul>
</li>
<li>Trimming methods
<ul>
<li><strong>Hard trimming</strong>: trim a fixed number of bases from the beginning and/or
end</li>
<li><strong>Quality trimming</strong>: cut the end of the read as soon as the base quality
drops below a threshold</li>
<li><strong>Soft trimming</strong>: trim the read that cannot be aligned to the reference
genome</li>
</ul>
</li>
</ul>
<h3 id="multiple-alignments-and-report-considerations"><a class="header" href="#multiple-alignments-and-report-considerations">Multiple alignments and report considerations</a></h3>
<ul>
<li>A read may have multiple valid alignments with identical or similarly good
alignment scores
<ul>
<li>Aligners may use different reporting strategies:
<ul>
<li><strong>Randomly</strong> select one alignment from the top-scoring alignments</li>
<li>Report <u>all</u> alignments that are <u>within delta</u> of the
top-scoring alignment; clip if more than $N_{max}$ alignments are found</li>
<li>Report only alignments if they are <strong>unique</strong> (no other alignment within
delta of the alignment score)</li>
<li>Do not report anything if more than $N_{max}$ valid alignments are found</li>
<li>…</li>
</ul>
</li>
</ul>
</li>
<li>Whether a read has a <strong>unique alignment</strong> depends on
<ul>
<li>the read sequence and the sequence homology of the organism</li>
<li>the search algorithm of the aligner</li>
</ul>
</li>
<li>Whether a read has a <strong>single reported alignment</strong> depends <em>additionally</em> on
<ul>
<li>the reporting options strategy of the aligner</li>
</ul>
</li>
</ul>
<h2 id="expression-quantification"><a class="header" href="#expression-quantification">Expression quantification</a></h2>
<ul>
<li>Expression quantification on
<ul>
<li><strong>Gene level</strong>
<ul>
<li>Reads belong to a gene locus</li>
</ul>
</li>
<li><strong>Isoform level</strong>
<ul>
<li>Reads belong to an isoform</li>
</ul>
</li>
</ul>
</li>
</ul>
<h3 id="how-to-count-reads"><a class="header" href="#how-to-count-reads">How to count reads?</a></h3>
<ul>
<li><strong>Unique mapped reads</strong>
<ul>
<li>Multi-reads are ignored – potentially biased quantification</li>
</ul>
</li>
<li><strong>All mapped reads</strong>
<ul>
<li>Assignment of multi-reads need abundance estimation of overlapped
genes/isoforms</li>
</ul>
</li>
</ul>
<p><img src="img/03-read-counting.png" alt="Read counting" /></p>
<h4 id="em-estimation"><a class="header" href="#em-estimation">EM estimation</a></h4>
<ul>
<li>General steps
<ol>
<li>Estimate abundances based on <u>uniquely mapping reads only</u></li>
<li>For each multi-read, divide it between the transcripts to which it maps, proportionally to their abundances estimated in the first step</li>
<li>Recompute abundances based on updated counts for each transcript
r.</li>
<li>Continue with Step 2</li>
</ol>
</li>
<li>Model the sequencing and analysis process with a likelihood function</li>
<li>E-step: Compute expected read counts given current expression levels</li>
<li>M-step: Compute expression values maximizing likelihood given expected read counts</li>
</ul>
<h4 id="alignment-free-rna-seq-quantification"><a class="header" href="#alignment-free-rna-seq-quantification">Alignment-free RNA-seq quantification</a></h4>
<p>Instead of building BWT, we build hashed k-mer index based on known
transcriptome, and search the k-mers in the reads to map and compute abundance.</p>
<p>Easier to understand through side-by-side comparison with the traditional
alignment dependent method:</p>
<p><img src="img/03-kmer.png" alt="Alignment dependent v.s. alignment free" /></p>
<p>While allocating k-mer to transcript, we try to make each transcript covered
as uniformly as possible.</p>
<ul>
<li>Pros
<ul>
<li>Accurate and fast in quantifying <strong>KNOWN TRANSCRIPTS</strong></li>
<li>Counts can be aggregated to get gene-level quantification</li>
</ul>
</li>
<li>Cons
<ul>
<li>Less well annotated genomes – less accurate results</li>
<li>No alignments – no information about SNPs</li>
<li>Transcript coverage is not uniform</li>
</ul>
</li>
</ul>
<h3 id="normalization"><a class="header" href="#normalization">Normalization</a></h3>

                    </main>

                    <nav class="nav-wrapper" aria-label="Page navigation">
                        <!-- Mobile navigation buttons -->
                            <a rel="prev" href="../../23fs/fg/02_modern_genomics_ii.html" class="mobile-nav-chapters previous" title="Previous chapter" aria-label="Previous chapter" aria-keyshortcuts="Left">
                                <i class="fa fa-angle-left"></i>
                            </a>

                            <a rel="next prefetch" href="../../23fs/fg/04_transcriptomics_ii.html" class="mobile-nav-chapters next" title="Next chapter" aria-label="Next chapter" aria-keyshortcuts="Right">
                                <i class="fa fa-angle-right"></i>
                            </a>

                        <div style="clear: both"></div>
                    </nav>
                </div>
            </div>

            <nav class="nav-wide-wrapper" aria-label="Page navigation">
                    <a rel="prev" href="../../23fs/fg/02_modern_genomics_ii.html" class="nav-chapters previous" title="Previous chapter" aria-label="Previous chapter" aria-keyshortcuts="Left">
                        <i class="fa fa-angle-left"></i>
                    </a>

                    <a rel="next prefetch" href="../../23fs/fg/04_transcriptomics_ii.html" class="nav-chapters next" title="Next chapter" aria-label="Next chapter" aria-keyshortcuts="Right">
                        <i class="fa fa-angle-right"></i>
                    </a>
            </nav>

        </div>




        <script>
            window.playground_copyable = true;
        </script>


        <script src="../../elasticlunr.min.js"></script>
        <script src="../../mark.min.js"></script>
        <script src="../../searcher.js"></script>

        <script src="../../clipboard.min.js"></script>
        <script src="../../highlight.js"></script>
        <script src="../../book.js"></script>

        <!-- Custom JS scripts -->
        <script src="../../mermaid.min.js"></script>
        <script src="../../mermaid-init.js"></script>


    </div>
    </body>
</html>
